rbi/core/enumerable.rbi

```diff
@@ -374,17 +374,19 @@ module Enumerable
   # (1..100).find    { |i| i % 5 == 0 and i % 7 == 0 }   #=> 35
   # ```
   sig do
-    params(
-        ifnone: Proc,
+    type_parameters(:U)
+      .params(
+        ifnone: T.proc.returns(T.type_parameter(:U)),
         blk: T.proc.params(arg0: Elem).returns(BasicObject),
-    )
-    .returns(T.nilable(Elem))
+      )
+      .returns(T.any(T.type_parameter(:U), Elem))
   end
   sig do
-    params(
-        ifnone: Proc,
-    )
-    .returns(T::Enumerator[Elem])
+    type_parameters(:U)
+      .params(
+        ifnone: T.proc.returns(T.type_parameter(:U)),
+      )
+      .returns(T::Enumerator[T.any(T.type_parameter(:U), Elem)])
```

Comment:
This raises some issues on the stripe codebase, as leaving off the `ifnone` block parameter means that the result will just be nilable. I think you probably just need a third overload that doesn't include it. Could you add a test for this case:

```
T.reveal_type([1,2].detect {|x| false}) # error: Revealed type `T.nilable(Integer)`
```

Reply:
Good catch. I fixed this and added a bunch of tests in `test/testdata/rbi/enumerable.rb` 👍